osquery/tables/system/windows/system_util.cpp

```diff
@@ -89,36 +89,95 @@ std::string WmiResultItem::GetString(const std::string& name) const {
   return BSTR_to_string(value.bstrVal);
 }
 
+// Returns a pair with the first value indicating success of the query
+// and the second value being the data obtained from the query.
+std::pair<bool, bool> WmiResultItem::GetBool(const std::string& name) const {
+  std::wstring property_name = string_to_wstring(name);
+
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+
+  if (hr != S_OK || value.vt != VT_BOOL) {
+    return std::make_pair(false, false);
+  }
+  return std::make_pair(true, value.boolVal);
+}
+
+std::vector<std::string> WmiResultItem::GetVectorOfStrings(
```

Comment:
On the note of the above, would it be better to have all of these WMI functions return a status, and pass in a reference to something they add to? So, each of these functions possibly takes a Row or something, adds data to that, and then returns a status?